git_repo/services/ext/gitbucket.py

```diff
@@ -0,0 +1,69 @@
+#!/usr/bin/env python
+
+import logging
+log = logging.getLogger('git_repo.gitbucket')
+
+from ..service import register_target
+from .github import GithubService
+
+import github3
+
+@register_target('bucket', 'gitbucket')
+class GitbucketService(GithubService):
+    fqdn = "localhost"
+    port = 8080
+    def __init__(self, *args, **kwarg):
+        super(GitbucketService, self).__init__(*args, **kwarg)
+
+    def connect(self):
+        self.gh = github3.GitHubEnterprise(self.build_url(self))
```

Comment:
...?
I talk about timing of loading settings from `.gitconfig`.

1. on `GitBucketService.__init__` calls `GithubService.__init__`
2. on `GithubService.__init__`, assign `self.gh` to `GitHubEnterprise` with `build_url`
3. on `GithubService.__init__`, calls `RepositoryService.__init__`
4. on `RepositoryService.__init__` calls `load_configuration`

`.gitconfig`'s fqdn/port settings load at 4. but `build_url` called at 2.

and even if rewrite `__init__` in `github.py` as:
```python
    def __init__(self, *args, **kwarg):
        super(GithubService, self).__init__(*args, **kwarg)```
        if self.fqdn == GITHUB_COM_FQDN:
            self.gh = github3.GitHub()
        else:
            self.gh = github3.GitHubEnterprise(RepositoryService.build_url(self))
```

it get error because `RepositoryService.__init__` calls `self.connect`. it uses `self.gh`.